### DISCUSSION TITLE
Hosting Multiple Mastodon media on one S3 bucket

### DISCUSSION THREAD

**Discussion by Pallav-Reactiveworks** [mastodon/mastodon#27904] [Q&A]
Hi,
issue: not able to host multiple mastodon media on one s3 bucket.

i can host one mastodon media_attachment on s3 bucket successfully with (s3://bucketname/). but when i'm trying to use one s3 bucket to host multiple mastodon media_attachment like(s3://mybucket/mastodon1/media_attachment.., s3://mybucket/mastodon2/media_attachment) in that case i'm not able to attache media at time of toots. getting 500 error.

**Comment by shleeable** [comment]
I believe this is failing because the S3 config doesn't support a prefix/sub folder. how is your mastodon env configured?

+ There was a PR somewhere with the intention of sharing media around multiple instances.. this was for places that hosted multiple instances on the same server, and it wouldn't transcode the same video multiple times... no idea if that's suitable in your case.

**Comment by weex** [comment]
Why not setup another bucket?

**Comment by S0yKaf** [comment]
I opened a pull request that adds a env var to prefix bucket keys
#30181

@weex 
>Why not setup another bucket?

Some object providers like backblaze limit the amount of buckets you can have on your account. It's also a lot more work to automate bucket creation versus just prefixing your keys and re-using the same bucket.
